oci/spec_opts.go

```diff
@@ -338,6 +339,24 @@ func WithImageConfig(image Image) SpecOpts {
 	return WithImageConfigArgs(image, nil)
 }
 
+// imageConfigExtended is an extension of the oci ImageConfig struct that adds
+// support for `ArgsEscaped`. This allows for proper support of the 'shell form'
+// of Entrypoint or Cmd from a Dockerfile for Windows containers. Please see
+// WithImageConfigArgs windows assignment for more information.
+type imageConfigExtended struct {
+	Entrypoint  []string `json:"Entrypoint,omitempty"`
+	Cmd         []string `json:"Cmd,omitempty"`
+	ArgsEscaped bool     `json:"ArgsEscaped,omitempty"`
+}
+
+// imageExtended is a subset of the oci Image struct that changes the Config
+// type to be imageConfigExtended in order to add the ability to deserialize
+// `ArgsEscaped` which is not an OCI field, but is supported by Docker built
+// images.
+type imageExtended struct {
+	Config imageConfigExtended `json:"config,omitempty"`
+}
+
 // WithImageConfigArgs configures the spec to from the configuration of an Image with additional args that
 // replaces the CMD of the image
 func WithImageConfigArgs(image Image, args []string) SpecOpts {
```

Comment:
It looks like `WithImageConfigArgs` is not used by CRI. I think we would need a similar change [here](https://github.com/containerd/containerd/blob/f032016af6bfcece41ed281a126cf9ba9badbb1c/pkg/cri/opts/spec.go#L54-L74) as well.